### PR TITLE
The cherrio config route

### LINKED ISSUE #349: 支持通过配置文件添加源
一个提议，即用模板生成新的源。
在自己的分支中实现过，不确定是否适用这里的需求。

1. [模板 template_page.js （我在自己分支中的一种实现）](https://github.com/josherich/RSSHub/blob/master/routes/template_page.js)

2. 配置写法的一个例子
```
const template = require('../template_page');

const options = {
    feed_title: '海关总署 - 动态',
    feed_desc: '海关总署 - 动态',
    feed_image: 'http://www.customs.gov.cn/customs/template/common/nav/logo.png',
    feed_url: 'http://www.customs.gov.cn/customs/xwfb34/302262/302263/index.html',
    url: 'http://www.customs.gov.cn/customs/xwfb34/302262/302263/index.html',
    baseUrl: 'http://www.customs.gov.cn/',
    list_slr: ['li', '.conList_ul'],
    title_slr: 'a',
    link_slr: 'a',
    desc_slr: 'a',
    time_slr: 'span',
    link_rel: true,
    cn: false,
};

module.exports = template(options);
```

### LINKED ISSUE #1035: 希望增加裏垢女子まとめ的rss
## 新 RSS 需求 | New RSS requirement

-   需求描述 | Description ：获得推文里的多张图片或视频，以及获得某个特定user的rss
-   是否已提供官网 RSS | Does the website provide an official RSS feed：No
-   源网站地址 | Website URL： https://www.uraaka-joshi.com ，或 某个特定user，如 https://www.uraaka-joshi.com/users/_rrwq/

feed43做出来的只能获得单张图片，多张限制条件太多又不能覆盖全部推文，视频抽不出来，寄希望于绅士大佬。

### PR DESCRIPTION
**Pull request by SettingDust**
Use js config too build rss

close #349 close #1035

### CODE REVIEW

File: docs/joinus/README.md [hunk 1]
```diff
@@ -397,7 +397,7 @@ ctx.state.data = {
 | 脚本路径                   | `./routes/github/issue`                                                      |
 | lib/router.js 中的完整代码 | `router.get('/github/issue/:user/:repo', require('./routes/github/issue'));` |
 
-3. [embassy](https://github.com/DIYgod/RSSHub/blob/master/lib/routes/embassy/index.js)
+3. [embassy](https://github.com/DIYgod/RSSHub/blob/master/lib/routes/embassy/common-config.js)
```
Comment (DIYgod) [inline docs/joinus/README.md]: 这是啥？

File: docs/en/joinus/README.md [hunk 2]
```diff
@@ -402,7 +402,7 @@ Add the script into [/lib/router.js](https://github.com/DIYgod/RSSHub/blob/maste
 | Route Path                         | `./routes/github/issue`                                                      |
 | the complete code in lib/router.js | `router.get('/github/issue/:user/:repo', require('./routes/github/issue'));` |
 
-3. [embassy](https://github.com/DIYgod/RSSHub/blob/master/lib/routes/embassy/index.js)
+3. [embassy](https://github.com/DIYgod/RSSHub/blob/master/lib/routes/embassy/common-config.js)
```
Comment (DIYgod) [inline docs/en/joinus/README.md]: 这是啥？

File: Procfile [hunk 3]
```diff
@@ -1 +1 @@
-web: node lib/index.js
+web: node lib/common-config.js
```
Comment (DIYgod) [inline Procfile]: 这是啥？

File: docs/README.md [hunk 4]
```diff
@@ -3110,3 +3110,9 @@ type 为 all 时，category 参数不支持 cost 和 free
 ### 决胜网
 
 <route name="最新资讯" author="WenryXu" example="/juesheng" path="/juesheng"/>
+
+### 裏垢女子まとめ
+
+<route name="主页" author="SettingDust"  example="/uraaka-joshi" path="/common/uraaka-joshi"/>
```
Comment (DIYgod) [inline docs/README.md]: path 不对

File: lib/utils/common-config.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+                    pubDate: getProp(data, ['item', 'pubDate'], $elem),
+                    link: getProp(data, ['item', 'link'], $elem),
+                    guid: getProp(data, ['item', 'guid'], $elem),
+                };
+            })
+            .get(),
+    };
+}
+
+module.exports = buildData;
```
Comment (DIYgod) [inline lib/utils/common-config.js]: 需要补充下[使用文档](https://docs.rsshub.app/joinus/#%E8%8E%B7%E5%8F%96%E6%BA%90%E6%95%B0%E6%8D%AE)

**Comment by DIYgod** [comment]
Merged, thanks!

File: lib/routes/uraaka-joshi/uraaka-joshi-user.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+    const link = `https://www.uraaka-joshi.com/users/${params.id}`;
+    ctx.state.data = await buildData({
+        link,
+        url: link,
+        title: `$('.top-profile-card-name-link').text() + '@${params.id} - 裏垢女子まとめ'`,
+        item: {
+            item: '.content-main .stream .stream-item',
+            title: `$('.post-name').text() + '@${params.id} - 裏垢女子まとめ'`,
+            link: `https://www.uraaka-joshi.com/users/${params.id}`,
+            description: `$('.post .context').html()`,
```
Comment (DerekTso) [inline lib/routes/uraaka-joshi/uraaka-joshi-user.js]: @SettingDust 
hi, it seems that we should handle the video item manually, not just parsing the html. The video tag misses the attribute "controls" and it won't show up at the rss reader ( e.g. feedly).

**Pull request merged**